src/sentry/coreapi.py

```diff
@@ -411,6 +411,10 @@ def should_filter(self, project, data, ip_address=None):
             if filter_obj.is_enabled() and filter_obj.test(data):
                 return (True, six.text_type(filter_obj.id))
 
+        if matches_discarded_hash(data, project):
+            metrics.incr('preprocess.discarded', instance=project.id)
```

Comment:
It'd be nice if this log message included the matching discarded hash (either the primary key or the hash value) for debugging